ducktape/tests/test.py

```diff
@@ -292,41 +365,7 @@ def test_name(self):
         return ".".join(filter(lambda x: x is not None and len(x) > 0, name_components))
 
     @property
-    def logger_name(self):
-        return self.test_id
-
-    def configure_logger(self):
-        """Set up the logger to log to stdout and files.
-        This creates a directory and a few files as a side-effect.
-        """
-        if self._logger_configured:
-            raise RuntimeError("test logger should only be configured once.")
-
-        self._logger.setLevel(logging.DEBUG)
-        mkdir_p(self.results_dir)
-
-        # Create info and debug level handlers to pipe to log files
-        info_fh = logging.FileHandler(os.path.join(self.results_dir, "test_log.info"))
-        debug_fh = logging.FileHandler(os.path.join(self.results_dir, "test_log.debug"))
-
-        info_fh.setLevel(logging.INFO)
-        debug_fh.setLevel(logging.DEBUG)
-
-        formatter = logging.Formatter(ConsoleDefaults.TEST_LOG_FORMATTER)
-        info_fh.setFormatter(formatter)
-        debug_fh.setFormatter(formatter)
-
-        self._logger.addHandler(info_fh)
-        self._logger.addHandler(debug_fh)
-
-        ch = logging.StreamHandler(sys.stdout)
-        ch.setFormatter(formatter)
-        if self.session_context.debug:
-            # If debug flag is set, pipe verbose test logging to stdout
-            ch.setLevel(logging.DEBUG)
-        else:
-            # default - pipe warning level logging to stdout
-            ch.setLevel(logging.WARNING)
-        self._logger.addHandler(ch)
-
-
+    def logger(self):
+        """Lazily instantiate logger"""
+        return test_logger(
```

Comment:
This isn't just lazy, it actually reinstantiates if on every call. I suspect you want to cache this once it is constructed.